### PR TITLE
Add ability to wait for cert packs to be active

### PR DESCRIPTION
**Pull request by ct-dh**
This is probably not ready for merge - I made this little patch to test the feature request in #1559 and won't be able to follow up in all likelihood by the time there is a response due to a change in employer (so I will no longer personally be working with Cloudflare - but the use case for the employer is still valid/relevant), but I didn't want to just delete the local code.

Please feel free to use/modify/drop this as needed.

It adds a new option to the ﻿﻿`cloudflare_certificate_pack` resource to block/wait on creation until all certificates in the pack are marked as active in the API - with the aim of this making use of `create_before_destroy` less likely to cause service disruption on updates.

### CODE REVIEW

File: cloudflare/resource_cloudflare_certificate_pack.go [hunk 1]
```diff
@@ -62,6 +63,32 @@ func resourceCloudflareCertificatePackCreate(d *schema.ResourceData, meta interf
...
+			if len(certificatePack.Certificates) == 0 {
+				return resource.RetryableError(fmt.Errorf("certificate list in response is empty"))
+			}
+			for _, certificate := range certificatePack.Certificates {
+				if certificate.Status != "active" {
+					return resource.RetryableError(fmt.Errorf("expected all certificates in certificate pack to be active state but certificate %s was in state %s", certificate.ID, certificate.Status))
+				}
+			}
+
+			// if we get here, the new cert pack is active
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_certificate_pack.go:81]: is there a reason we're duplicating the `d.SetId` and `Read` operations here? we should be able to reuse what we already have a few lines further down.

**Comment by jacobbednarz** [comment]
before we consider merging this, we'll also need some tests and to clean up the imports for these files.

File: cloudflare/resource_cloudflare_certificate_pack.go (hunk 1 above)
Comment (ct-dh) [reply cloudflare/resource_cloudflare_certificate_pack.go]: No good reason, I had rushed this out so just hadn't noticed the duplication. Should be resolved in latest commits.

Comment (ct-dh) [reply cloudflare/resource_cloudflare_certificate_pack.go]: No good reason, I had rushed this out so just hadn't noticed the duplication. Should be resolved in latest commits.

**Comment by jacobbednarz** [comment]
> I can add a `resource.TestCheckResourceAttr` check for the new attribute to check the default

yep, just a check that it ends up in the state is fine. we can also add another test with the `wait_for_active_status` flag set to on so that we exercise that code path and there aren't any errors too.

**Comment by ct-dh** [comment]
Added the tests and rebased from master if you can take another look please @jacobbednarz.

I don't have a test Cloudflare account that would be safe to run the test suite in (from a brief look at the way they work, the sweepers will destroy unrelated resources) so I don't know if the tests actually pass or not.

**Comment by jacobbednarz** [comment]
acceptance tests look good

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCertificatePack_" -count 1 -parallel 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCertificatePack_AdvancedDigicert
--- PASS: TestAccCertificatePack_AdvancedDigicert (9.51s)
=== RUN   TestAccCertificatePack_AdvancedLetsEncrypt
--- PASS: TestAccCertificatePack_AdvancedLetsEncrypt (9.86s)
=== RUN   TestAccCertificatePack_DedicatedCustom
    resource_cloudflare_certificate_pack_test.go:133: Pending investigation into ACM entitlements
--- SKIP: TestAccCertificatePack_DedicatedCustom (0.00s)
=== RUN   TestAccCertificatePack_WaitForActive
--- PASS: TestAccCertificatePack_WaitForActive (14.06s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/cloudflare	34.009s
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/maintainer-only-file-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/tf-log-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
```

**Pull request merged**